integration-cli/docker_cli_pull_test.go

```diff
@@ -59,28 +59,36 @@ func TestPullVerified(t *testing.T) {
 	// unless keychain is manually updated to contain the daemon's sign key.
 
 	verifiedName := "hello-world"
+	expected := "The image you are pulling has been verified"
+	untargeted := "v1"
+
 	defer deleteImages(verifiedName)
 
 	// pull it
-	expected := "The image you are pulling has been verified"
 	pullCmd := exec.Command(dockerBinary, "pull", verifiedName)
-	if out, exitCode, err := runCommandWithOutput(pullCmd); err != nil || !strings.Contains(out, expected) {
-		if err != nil || exitCode != 0 {
-			t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
-		}
-		t.Fatalf("pulling a verified image failed. expected: %s\ngot: %s, %v", expected, out, err)
+	out, exitCode, err := runCommandWithOutput(pullCmd)
+	if err != nil || exitCode != 0 {
+		t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
 	}
 
-	// pull it again
-	pullCmd = exec.Command(dockerBinary, "pull", verifiedName)
-	if out, exitCode, err := runCommandWithOutput(pullCmd); err != nil || strings.Contains(out, expected) {
-		if err != nil || exitCode != 0 {
-			t.Skipf("pulling the '%s' image from the registry has failed: %s", verifiedName, err)
+	if strings.Contains(out, untargeted) {
+		logDone("pulling from v1 registry, no need verified")
```

Comment:
It probably depends on how you look at the test.

I'm ignoring the fact that the name of the test has the word "verify" in it and I look at this test as "make sure that doing a pull works correctly".  If that pull hits a v2 registry then it should verify the image.  If it hits a v1 registry then it doesn't verify it - but we still make sure that the pull did what it was supposed to do.

If you look at the test as "make sure verification works and therefore only check stuff if it hits a v2 registry", then I can understand why someone would think doing a skip() make sense - you can't test what you wanted to.  But even in that case I don't see the harm in at least checking what we can check.  But if @dmcgowan feels strongly that its misleading to not skip then that's fine.